doc/md/writing-motoko/pattern-matching.md

```diff
@@ -100,6 +100,22 @@ Some types contain just a single value. We call these singleton types. Examples
 
 ### Exhaustiveness (coverage) checking
 
-
 At runtime, a switch expression may wind up scrutinizing a value to which none of its alternative patterns apply, generating an undesired trap.
-To detect the possibility of such runtime failures, the Motoko compiler checks for the exhaustiveness of pattern matching by keeping track of the covered shape of the scrutinee. The compiler issues a warning for any non-covered scrutinees. Motoko even constructs a helpful example of a scrutinee that is not matched. A useful by-product of the exhaustiveness check is that it identifies and warns about dead or redundant alternatives that can never be matched.
\ No newline at end of file
+To detect the possibility of such runtime failures, the Motoko compiler checks for the exhaustiveness of pattern matching by keeping track of the covered shape of the scrutinee. The compiler issues a warning for any non-covered scrutinees. Motoko even constructs a helpful example of a scrutinee that is not matched. A useful by-product of the exhaustiveness check is that it identifies and warns the developer about dead or redundant alternatives that can never be matched.
+
+## Refutable patterns and dealing with non-matching data
+
+Developers may only be interested in specially formed data with a desire to directly handle all non-matching forms. The `let`-`else` construct is designed precisely for this purpose. Whereas the regular destructuring `let` allows to focus on a single given pattern, it invariably traps if the right-hand side data doesn't match it, which is warned at compilation. The `else` clause gives the programmer a way to deal with refuted matches, such as bailing out of the process or logging a message before trapping. As such, `let`-`else` is similar to a two-`case` `switch` in a compact form that additionally doesn't force the indentation of the processing logic following it.
+
+The below example illustrates how you can write a non-indenting `if`-`else` by resorting to a `let`-`else` in your code:
+
+``` motoko
+let true = isLoggedIn(customer) else return;
+// process message for logged-in customer
+```
```

Comment:
This example isn't very motivating since you could just use 
```
is (!IsLoggedIn(customer)) return';
... 
```
What about something that binds some identifiers? Along the lines of (unchecked):
```
let (?record) = map.get(map, customer) else return Error.error("no such customer")
````

Reply:
Yeah, this is "parse, don't validate". Good point.

Reply:
I'll  use this from Claude...

``` motoko
func getName(optionalName : ?Text) : Text {
  let ?name = optionalName else return "Unknown";
  name
}
```